org.eclipse.jface.text/src/org/eclipse/jface/text/link/LinkedModeUI.java

```diff
@@ -279,14 +279,40 @@ public interface IExitPolicy {
 		ExitFlags doExit(LinkedModeModel model, VerifyEvent event, int offset, int length);
 	}
 
+	/**
+	 * An extension to exit policy can be registered by a caller to get custom exit behavior for
+	 * external changes like Copy-Paste operations.
+	 * <p>
+	 * Clients may implement this interface.
+	 * </p>
+	 *
+	 * @since 3.21
+	 */
+	public interface IExitPolicyExt extends IExitPolicy {
```

Comment:
Instead of an `Ext` interface, we can now add `default` methods on existing interfaces and this is backward compatible. Can you please rework it to use that?
Also, it would be great if this could come with an automated test.

Reply:
> Adding a method into an exiting interface requires us to bump Major version for the plugin

Try to add "default" method.

Reply:
@iloveeclipse I did it as @mickaelistria suggested... You can spot it in the error message:
```
The default method org.eclipse.jface.text.link.LinkedModeUI.IExitPolicy.doExit(LinkedModeModel, DocumentEvent) in an interface has been added
```
It doesn't matter what kind of method is added to an existing interface ("default"  or not) - it's all treated as API break and, as such, requires to bump the major version.

Reply:
Would you please provide alternative patch with default method ? If what you say happens this is a bug in PDE and we should add api filter.